packages/calcite-components/src/components/modal/modal.tsx

```diff
@@ -329,6 +334,19 @@ export class Modal
 
   // #region Private Methods
 
+  private keyDownHandler = (event: KeyboardEvent): void => {
+    const { defaultPrevented, key } = event;
+
+    if (defaultPrevented) {
```

Comment:
This should also return when focus trap is enabled. Applies to similar event handlers.